### PR TITLE
Add grad scaler

### PR DESCRIPTION
**Pull request by andersonic**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: benchmarks/pipe.py [hunk 1]
```diff
@@ -9,9 +9,10 @@
 from torchtext.data.utils import get_tokenizer
```
Comment (shruti-bh) [inline benchmarks/pipe.py:9]: why did you rename the file to pipe.py?

**Review by shruti-bh — APPROVED** [review]

File: benchmarks/pipe.py (hunk 1 above)
Comment (andersonic) [reply benchmarks/pipe.py:9]: I feel like the aspect of Fairscale that it's benchmarking is primarily pipe (though it should also detect regressions in purefp16)
